File: data_structure/linkedlist/go/linkedlist.go
package linkedlist

import "fmt"

type Node struct {
	value int
	next  *Node
}

type Linkedlist struct {
	head   *Node
	length int
}

func (l *Linkedlist) InsertAtBeggning(val int) {
	n := Node{}
	n.value = val
	n.next = l.head
	l.head = &n
	l.length++
}

func (l *Linkedlist) Print() {
	if l.length == 0 {
		fmt.Println("No nodes in list")
	}
	ptr := l.head
	for i := 0; i < l.length; i++ {
		fmt.Println("Node: ", ptr)
		ptr = ptr.next
	}
}

File: data_structure/linkedlist/js/linkedlist.js
class Node {

  constructor(val, next = null) {
    this.value = val
    this.next = next
  }

}

class Linkedlist {
  constructor(head = null) {
    this.head = head
    this.length = 0
  }

  setNewHead(head) {
    this.head = head
  }

  InsertAtBeggning(val) {
    let node = new Node(val,this.head)
    this.head = node
    this.length++
  }

  Print() {
    if (this.length == 0) {
      console.log("No nodes in list")
    }
    let ptr = this.head
    for (let i = 0; i < this.length; i++) {
      console.log("Node: ", ptr)
      ptr = ptr.next
    }
  }
}

module.exports = {
  Linkedlist
}
